base/commands/migration/cancel_it_test.go

```diff
@@ -34,30 +35,36 @@ func noMigrationsCancelTest(t *testing.T) {
 	tcx := it.TestContext{T: t}
 	ctx := context.Background()
 	tcx.Tester(func(tcx it.TestContext) {
+		createMapping(ctx, tcx)
 		err := tcx.CLC().Execute(ctx, "cancel")
-		require.Contains(t, err.Error(), "there are no migrations in progress")
+		require.Contains(t, err.Error(), "finding migration in progress: no rows found")
 	})
 }
 
 func cancelTest(t *testing.T) {
 	tcx := it.TestContext{T: t}
 	ctx := context.Background()
 	tcx.Tester(func(tcx it.TestContext) {
-		mID := migration.MakeMigrationID()
+		mID := migrationIDFunc()
 		createMapping(ctx, tcx)
-		statusMap := MustValue(tcx.Client.GetMap(ctx, migration.StatusMapName))
-		b := MustValue(os.ReadFile("testdata/cancel/migration_cancelling.json"))
-		Must(statusMap.Set(ctx, mID, serialization.JSON(b)))
-		l := MustValue(tcx.Client.GetList(ctx, migration.MigrationsInProgressList))
-		m := MustValue(json.Marshal(migration.MigrationInProgress{
-			MigrationID: mID,
-		}))
-		ok := MustValue(l.Add(ctx, serialization.JSON(m)))
-		require.Equal(t, true, ok)
-		tcx.WithReset(func() {
+		setStatusInProgress(tcx, ctx)
+		var wg sync.WaitGroup
+		wg.Add(1)
+		go tcx.WithReset(func() {
+			defer wg.Done()
 			Must(tcx.CLC().Execute(ctx, "cancel"))
 		})
-		MustValue(l.Remove(ctx, serialization.JSON(m)))
+		time.Sleep(1 * time.Second)
```

Comment:
We may have a problem if these tests don't work without sleeping.

Reply:
I could not find a way to work them without sleeping, because it is a simulation of changing states. State changes does not depend on any condition that eventually does not help. So we cannot simulate the behaviour in any other way. I debugged these tests and without sleeping it immediately changes the state to latest state that commands cannot process the initial states which makes the tests fail. Sorry, I looked for it for a while and I have no other solution for this. Also I think we don't introduce flaky tests, because 1 second it a good enough time a command to process the initial state on any machine IMO.

Reply:
(also I don't get why we have sleep in this test, how mocking of migration cluster is done is complex for me. )

Reply:
https://github.com/hazelcast/hazelcast-commandline-client/pull/424/commits/0ba87e67479948d8b7dff6fd94a9a179d50eed57